inst/include/dust/filter_state.hpp

```diff
@@ -133,18 +133,25 @@ class filter_trajectories_host {
     for (size_t i = 0; i < n_particles_; ++i) {
       index_particle[i] = i;
     }
-    for (size_t k = 0; k < n_data_ + 1; ++k) {
-      size_t i = n_data_ - k;
+    const size_t n_state_particles = n_state_ * n_particles_;
+    for (size_t k = 0; k < offset_; ++k) {
+      size_t i = offset_ - k - 1;
       auto const it_order = order_begin + i * n_particles_;
-      auto const it_value = value_begin + i * n_state_ * n_particles_;
-      auto it_ret = ret + i * n_state_ * n_particles_;
+      auto const it_value = value_begin + i * n_state_particles;
+      auto it_ret = ret + i * n_state_particles;
       for (size_t j = 0; j < n_particles_; ++j) {
         const size_t idx = *(it_order + index_particle[j]);
         index_particle[j] = idx;
         std::copy_n(it_value + idx * n_state_, n_state_,
                     it_ret + j * n_state_);
       }
     }
+
+    if (offset_ < n_data_) {
```

Comment:
Comment on why this is needed/when it is used?

Reply:
added a comment on that